File: backend/src/main/java/com/inventory/entity/RawMaterial.java
package com.inventory.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "raw_materials")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class RawMaterial {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @NotBlank(message = "Raw material name is required")
    @Column(nullable = false)
    private String name;
    
    @NotNull(message = "Stock quantity is required")
    @PositiveOrZero(message = "Stock quantity must be zero or positive")
    @Column(nullable = false)
    private Integer stockQuantity;
    
    @OneToMany(mappedBy = "rawMaterial", cascade = CascadeType.ALL)
    private List<ProductRawMaterial> productRawMaterials = new ArrayList<>();
}

File: backend/src/main/kotlin/com/inventory/entity/RawMaterial.kt
package com.inventory.entity

import jakarta.persistence.*
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull
import jakarta.validation.constraints.PositiveOrZero

@Entity
@Table(name = "raw_materials")
data class RawMaterial(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null,
    
    @field:NotBlank(message = "Raw material name is required")
    @Column(nullable = false)
    var name: String = "",
    
    @field:NotNull(message = "Stock quantity is required")
    @field:PositiveOrZero(message = "Stock quantity must be zero or positive")
    @Column(nullable = false)
    var stockQuantity: Int = 0,
    
    @OneToMany(mappedBy = "rawMaterial", cascade = [CascadeType.ALL])
    var productRawMaterials: MutableList<ProductRawMaterial> = mutableListOf()
)
